src/renderer/PresetsBar.tsx

```diff
@@ -113,29 +114,37 @@ const PresetsBar = () => {
     'Jazz',
   ]);
 
-  const handleCreatePreset = (prev: string[]) => {
-    if (prev.indexOf(presetName) !== -1) {
-      // TODO: overwrite preset values for the existing preset
-    } else {
-      // TODO: add code for creating a preset
-      setSelectedPresetName(presetName);
+  const handleCreatePreset = async (prev: string[]) => {
+    await savePreset(presetName);
+
+    // If we are creating a new preset and not just updating an existing one
+    if (prev.indexOf(presetName) === -1) {
+      await setSelectedPresetName(presetName);
       const newPresets = [...prev, presetName].sort();
-      setPresetNames(newPresets);
+      await setPresetNames(newPresets);
     }
+
+    console.log('Finished Handling Create Preset!');
   };
 
-  const handleLoadPreset = () => {
-    // TODO: add code for loading a preset
+  const handleLoadPreset = async () => {
+    if (selectedPresetName) {
+      try {
+        await loadPreset(selectedPresetName);
+        await performHealthCheck();
+      } catch (error) {
+        console.log(`failed to get preset ${selectedPresetName}`);
+        console.log(error);
+      }
+    }
+    console.log('done loading preset!');
   };
 
   const handleChangeNewPresetName = (newValue: string) => {
     setPresetName(newValue);
 
-    // Update selectedPresetName if the input value matches an existing preset name
-    const newSelectedName = presetNames.some((n) => n === newValue)
-      ? newValue
-      : undefined;
-    setSelectedPresetName(newSelectedName);
+    // TODO: If the preset name is an existing one, we should set the selected one to be that so we can load it
+    setSelectedPresetName(undefined);
```

Comment:
Could we undo this change? It was a fix I made in jat-90 that I force-pushed so it likely didn't make it into this branch.